env/env.cc

```diff
@@ -22,6 +23,20 @@ namespace rocksdb {
 Env::~Env() {
 }
 
+Status Env::NewLogger(const std::string& fname,
+                      std::shared_ptr<Logger>* result) {
+  EnvOptions options;
+  std::unique_ptr<WritableFile> writable_file;
+  const auto status = NewWritableFile(fname, &writable_file, options);
+  if (!status.ok()) {
+    return status;
+  }
+
+  *result = std::make_shared<EnvLogger>(std::move(writable_file), fname,
+                                        options, this);
+  return Status::OK();
+}
```

Comment:
Maybe I missed something but I don't see this function is covered in unit test.

Reply:
This function can not be directly used in the test as this is a virtual method. PosixEnv which is being used in the test has overridden this method.

As you suggested, i have created a new public helper method NewEnvLogger(). This method is now being called from test.